src/main/scala/dsls/regex/RegularExpression.scala

```diff
@@ -11,26 +14,83 @@ package dsls.regex
 abstract class RegularExpression {
   /** returns true if the given string matches this regular expression */
   def matches(string: String) = RegexMatcher.matches(string, this)
+  def ||(other: RegularExpression) = Union(this, other)
+  def ~(other: RegularExpression) = Concat(this, other)
+  def <*> = Star(this)
+  def <+> = Concat(this, Star(this))
+
+  // for making a {n} method I was very confused how to do it 
+  // I was trying to create a different Literal class that could
+  // accept two parameters (but would give a defaut value for n)
+  // but realized that didn't seem right. I also tried making an implicit
+  // method that would take a char and a num or a literal and a num and return
+  // the proper RegularExpression. I couldnt get this to work. Eventually stack
+  // overflow http://stackoverflow.com/questions/3551366/can-i-overload-paranthesis-in-scala
+  // led me to the apply method in order to overload the braces
+  // I am not sure if this is the proper way but it passed my very singular
+  // test. (when I did for all n: Int, the loop took too long to run so its
+  // a simple test.)
+  def apply(num: Int): RegularExpression = {
+    var n: RegularExpression = EPSILON
+    var count = num
+    while(count > 0){
+        n = Concat(this, n)
+        count = count - 1
+    }
+    n
+  }
+  // {n} will need to use the curried function maybe
+
 }
 
+
 /** a regular expression that matches nothing */
-object EMPTY extends RegularExpression
+object EMPTY extends RegularExpression 
 
 /** a regular expression that matches the empty string */
-object EPSILON extends RegularExpression
+object EPSILON extends RegularExpression 
 
 /** a regular expression that matches a literal character */
-case class Literal(val literal: Char) extends RegularExpression
+case class Literal(literal: Char) extends RegularExpression 
+
+
+// or a char and num to literal
+
+// Literal Companion class for implicitConversions
+object LiteralImplicits {
+    implicit def char2literal(literal: Char): Literal = Literal(literal)
+
+    implicit def string2regexp(literal: String): RegularExpression = {
+        if (literal == ' ')
+            EPSILON
+        else
+            Concat(Literal(literal.charAt(0)), string2regexp(literal.substring(1)))
```

Comment:
Nice, much more elegant than my implementation.  This one tripped me up a bit and I didn't quite like my implementation where I split the string into a list of chars, and used foldLeft to Concat the elements.  This is much nicer.